src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1371,6 +1371,29 @@ def clear_traffic_routing(cmd, resource_group_name, name):
     set_traffic_routing(cmd, resource_group_name, name, [])
 
 
+def add_cors(cmd, resource_group_name, name, allowed_origins):
```

Comment:
Need to support all these commands for slots as well.

Reply:
Want to confirm it is supported with slot. I thought about that, but in portal there is no UI gesture for slot. Anyway, if yes, i will make the change

Reply:
Yes it is supported on slots for both: on the portal if you go to Deployment Slots you will see CORS in the menu blade. & supported on Functions as well
![image](https://user-images.githubusercontent.com/13839372/44111174-91f2073e-9fb6-11e8-80da-5f61d05405bf.png)

Reply:
Ah, got it I only tried out the webapp with slots, and the tab of "API->CORS" doesn't have the drop down for me to tweak the slot specific origins. Funcapp's ui makes more sense.

Reply:
Done for webapp. Note `deployment slot` as a feature, is not yet exposed for functionapp, so even though my change has already made cors capable for functionapp slot, but for consistency we should not make it visible till we enable `az functionapp deployment slot`. This way users would have full e2e under `az functionapp`. Sounds reasonable?
Also, when slot support for functionapp will default to be enabled?